packages/core/src/impl/QueryApiImpl.ts

```diff
@@ -23,27 +27,69 @@ export class QueryApiImpl implements QueryApi {
     return this
   }
 
+  lines(query: string, variables?: Variables): Observable<string> {
+    return new ObservableQuery(this.createExecutor(query, variables), identity)
+  }
+
+  rows(query: string, variables?: Variables): Observable<Row> {
+    return new ObservableQuery(
+      this.createExecutor(query, variables),
+      observer => {
+        return toLineObserver({
+          next(values, tableMeta) {
+            observer.next({values, tableMeta})
+          },
+          error(e) {
+            observer.error(e)
+          },
+          complete() {
+            observer.complete()
+          },
+        })
+      }
+    )
+  }
+
   queryLines(query: string, consumer: CommunicationObserver<string>): void {
+    this.execute(query, undefined, this.options, consumer)
+  }
+
+  queryRows(query: string, consumer: FluxResultObserver<string[]>): void {
+    this.execute(query, undefined, this.options, toLineObserver(consumer))
+  }
+
+  private execute(
+    query: string,
+    extern: unknown | undefined,
+    {org, type, gzip}: QueryOptions,
```

Comment:
What is the benefit of having QueryOptions parameter herein? We could destructure this.options in the fn body.

Reply:
the options of an observable query should not change after its creation, thats why the `createExecutor` copies the current options and then passes this copy to `execute`